apps/hannk/halide/convolution_generator.cpp

```diff
@@ -190,6 +190,9 @@ class Convolution : public Generator<Convolution> {
         // things computed at the tile to have constant size. We can't assume the
         // output is bigger than a minimum size. So, we specialize for decreasing
         // tile sizes, and have a degenerate tile case to handle the rest.
+        //
+        // TODO: this is actually faster in some cases if we just use vector_size=8
+        // (even on AVX2 / Skylake). Should investigate further.
```

Comment:
I realized why this is: on Intel, vector_reduction is 2, so vector_size is too big for the edge case. This is fine for now, it will take some more thought/specializations to get this right for the small number of channels output case.